internal/pkg/service/buffer/api/dependencies/dependencies.go

```diff
@@ -1,240 +0,0 @@
-	FileManager() *file.Manager
-}
-
-// forServer implements ForServer interface.
-type forServer struct {
-	serviceDependencies.ForService
-	config  config.Config
-	stats   *statistics.CollectorNode
-	watcher *watcher.APINode
-}
-
-// forPublicRequest implements ForPublicRequest interface.
-type forPublicRequest struct {
-	ForServer
-	logger     log.Logger
-	request    *http.Request
-	requestCtx context.Context
-	requestID  string
-}
-
-// forProjectRequest implements ForProjectRequest interface.
-type forProjectRequest struct {
-	dependencies.Project
-	ForPublicRequest
-	logger       log.Logger
-	tokenManager *token.Manager
-	tableManager *table.Manager
-	taskNode     *task.Node
-	fileManager  *file.Manager
-}
-
-func NewServerDeps(ctx context.Context, proc *servicectx.Process, cfg config.Config, envs env.Provider, logger log.Logger, tel telemetry.Telemetry) (v ForServer, err error) {
-	ctx, span := tel.Tracer().Start(ctx, "keboola.go.buffer.api.dependencies.NewServerDeps")
-	defer span.End(&err)
-
-	// Create service dependencies
-	userAgent := "keboola-buffer-api"
-	serviceDeps, err := serviceDependencies.NewServiceDeps(ctx, proc, cfg.ServiceConfig, envs, logger, tel, userAgent)
-	if err != nil {
-		return nil, err
-	}
-
-	// Create server dependencies
-	d := &forServer{
-		ForService: serviceDeps,
-		config:     cfg,
-	}
-
-	d.stats = statistics.NewCollectorNode(d)
-
-	d.watcher, err = watcher.NewAPINode(d)
-	if err != nil {
-		return nil, err
-	}
-
-	return d, nil
-}
-
-func NewDepsForPublicRequest(serverDeps ForServer, req *http.Request) ForPublicRequest {
-	ctx, span := serverDeps.Telemetry().Tracer().Start(req.Context(), "keboola.go.buffer.api.dependencies.NewDepsForPublicRequest")
-	defer span.End(nil)
-
-	requestId, _ := ctx.Value(middleware.RequestIDCtxKey).(string)
-
-	return &forPublicRequest{
-		ForServer:  serverDeps,
-		logger:     serverDeps.Logger().AddPrefix(fmt.Sprintf("[requestId=%s]", requestId)),
-		request:    req,
-		requestCtx: req.Context(),
-		requestID:  requestId,
-	}
-}
-
-func NewDepsForProjectRequest(publicDeps ForPublicRequest, ctx context.Context, tokenStr string) (v ForProjectRequest, err error) {
-	ctx, span := publicDeps.Telemetry().Tracer().Start(ctx, "keboola.go.buffer.api.dependencies.NewDepsForProjectRequest")
-	defer span.End(&err)
-
-	projectDeps, err := dependencies.NewProjectDeps(ctx, publicDeps, publicDeps, tokenStr)
-	if err != nil {
-		return nil, err
-	}
-
-	logger := publicDeps.Logger().AddPrefix(
-		fmt.Sprintf("[project=%d][token=%s]", projectDeps.ProjectID(), projectDeps.StorageAPITokenID()),
-	)
-
-	d := &forProjectRequest{
-		Project:          projectDeps,
-		ForPublicRequest: publicDeps,
-		logger:           logger,
-	}
-	d.tokenManager = token.NewManager(d)
-	d.tableManager = table.NewManager(d.KeboolaProjectAPI())
-	d.fileManager = file.NewManager(d.Clock(), d.KeboolaProjectAPI(), nil)
-	d.taskNode, err = task.NewNode(d)
-	if err != nil {
-		return nil, err
-	}
-	return d, nil
-}
```

Comment:
:lady_beetle: tu bol bug `d.taskNode, err = task.NewNode(d)`:
- Tj. pre kazdy request authentikovany tokenom sa vytvarala nova `task.Node` + nova etcd session.
  - Nemalo to vplyv na funkcnost ale na performance ano + neporiadok v logoch.

Reply:
Teraz tam uz tato chyba nie je:

https://github.com/keboola/keboola-as-code/blob/315ef855a88c42758de9a142493359712f74b868/internal/pkg/service/buffer/dependencies/dependencies.go#L79-L85

https://github.com/keboola/keboola-as-code/blob/315ef855a88c42758de9a142493359712f74b868/internal/pkg/service/buffer/dependencies/reqproject.go#L36-L53